packages/commonwealth/client/scripts/views/pages/contracts/index.tsx

```diff
@@ -29,6 +70,58 @@ const ContractsPage = () => {
   const onUpdateSuccess = () => {
     const updatedContracts = app.contracts.store.getCommunityContracts();
     setContracts([...updatedContracts]);
+    fetchTemplates();
+  };
+
+  const handleDeleteTemplate = (template) => {
+    if (template.inUse) {
+      openConfirmation({
+        title: 'Action template cannot be deleted',
+        description: (
+          <>
+            Action template <b>{template.name}</b> cannot be deleted at this
+            time because it is currently being used by a community.
+          </>
+        ),
+        buttons: [
+          {
+            label: 'Close',
+            buttonType: 'secondary-black',
+          },
+        ],
+      });
+    } else {
+      openConfirmation({
+        title: 'Delete Template',
+        description: (
+          <>
+            Deleting this template <b>{template.name}</b> is permanent and
+            deletes all associated data. Are you sure you want to proceed?
+          </>
+        ),
+        buttons: [
+          {
+            label: 'Delete',
+            buttonType: 'mini-red',
+            onClick: async () => {
+              try {
+                await app.contracts.deleteTemplate({
+                  templateId: template.id,
+                });
+                await fetchTemplates();
+              } catch (e) {
+                console.error(e);
+                notifyError('Failed to delete template!');
+              }
+            },
+          },
+          {
+            label: 'Cancel',
+            buttonType: 'mini-black',
+          },
+        ],
+      });
+    }
   };
 
   return (
```

Comment:
this return is massive and needs some decluttering as it is hard to see what is going on here. Avoid nested ternary